scripts/generate_duplicate_sheet.py

```diff
@@ -1,53 +1,116 @@
 from bugbug.models.duplicate import DuplicateModel
+from bugbug.utils import download_check_etag, zstd_decompress
+
+basicConfig(level=INFO)
+logger = getLogger(__name__)
 
-m = DuplicateModel.load("duplicatemodel")
+URL = "https://community-tc.services.mozilla.com/api/index/v1/task/project.relman.bugbug.train_similarity.latest/artifacts/public/{}.zst"
 
 REPORTERS_TO_IGNORE = {"intermittent-bug-filer@mozilla.bugs", "wptsync@mozilla.bugs"}
 
 
-try:
-    with open("duplicate_test_bugs.json", "r") as f:
-        test_bugs = json.load(f)
-except FileNotFoundError:
-    test_bug_ids = bugzilla.get_ids_between(
-        datetime.now() - timedelta(days=21), datetime.now()
-    )
-    test_bugs = bugzilla.get(test_bug_ids)
-    test_bugs = [
-        bug for bug in test_bugs.values() if not bug["creator"] in REPORTERS_TO_IGNORE
-    ]
-    with open("duplicate_test_bugs.json", "w") as f:
-        json.dump(test_bugs, f)
-
-bug_tuples = list(itertools.combinations(test_bugs, 2))
-
-# Warning: Classifying all the test bugs takes a while
-probs = m.classify(bug_tuples, probabilities=True)
-
-with open("duplicate_predictions.csv", "w") as csvfile:
-    spamwriter = csv.writer(csvfile)
-
-    spamwriter.writerow(
-        ["bug 1 ID", "bug 1 summary", "bug 2 ID", "bug 2 summary", "prediction"]
-    )
-
-    for bug_tuple, prob in zip(bug_tuples, probs):
-
-        if prob[1] > 0.8:
-            spamwriter.writerow(
-                [
-                    f'https://bugzilla.mozilla.org/show_bug.cgi?id={bug_tuple[0]["id"]}',
-                    bug_tuple[0]["summary"],
-                    f'https://bugzilla.mozilla.org/show_bug.cgi?id={bug_tuple[1]["id"]}',
-                    bug_tuple[1]["summary"],
-                    prob[1],
-                ]
+def parse_args(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--similaritymodel", default=None)
+    return parser.parse_args(args)
+
+
+def main(args):
+    if args.similaritymodel:
+        if args.similaritymodel == "elasticsearch":
+            similarity_model = similarity.model_name_to_class[args.similaritymodel]()
+        else:
+            model_file_name = f"{similarity.model_name_to_class[args.similaritymodel].__name__.lower()}.similaritymodel"
+
+            if not os.path.exists(model_file_name):
+                logger.info(
+                    f"{model_file_name} does not exist. Downloading the model...."
+                )
+                try:
+                    download_check_etag(URL.format(model_file_name))
+                except requests.HTTPError:
+                    logger.error(
+                        f"A pre-trained model is not available, you will need to train it yourself using the trainer script"
+                    )
+                    raise SystemExit(1)
+
+                zstd_decompress(model_file_name)
+                assert os.path.exists(
+                    model_file_name
+                ), "Decompressed file doesn't exist"
```

Comment:
Could you try using the `download_model` or `download_and_load_model` helper functions from bugbug/utils.py?